### PR TITLE
Fixed replay playback on local games

### PR DESCRIPTION
**Pull request by Mailaender**
Followup of https://github.com/OpenRA/OpenRA/pull/17903.

### CODE REVIEW

File: OpenRA.Game/Network/ReplayConnection.cs [hunk 1]
```diff
@@ -102,6 +104,8 @@ public ReplayConnection(string replayFilename)
 					}
 				}
 
+				latency = Math.Max(1, LobbyInfo.GlobalSettings.OrderLatency);
```
Comment (adammitchelldev) [inline OpenRA.Game/Network/ReplayConnection.cs:107]: Could you try this just always set to 1 and see if it breaks multiplayer replays? It is kind of silly that it was ever set to anything else.

Comment (Mailaender) [reply OpenRA.Game/Network/ReplayConnection.cs:107]: Nope, doesn't work:

```
Exception of type `System.IO.EndOfStreamException`: Attempted to read past the end of the stream.
  at OpenRA.StreamExts.ReadBytes (System.IO.Stream s, System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00021] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.StreamExts.ReadBytes (System.IO.Stream s, System.Int32 count) [0x0001b] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Network.ReplayConnection..ctor (System.String replayFilename) [0x00074] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Game.JoinReplay (System.String replayFile) [0x00005] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Mods.Common.Widgets.Logic.ReplayBrowserLogic.WatchReplay () [0x0002a] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.Logic.ReplayBrowserLogic+<>c__DisplayClass14_0.<.ctor>b__3 () [0x00000] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.ButtonWidget.<.ctor>b__42_5 (OpenRA.MouseInput _) [0x00000] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.ButtonWidget.HandleMouseInput (OpenRA.MouseInput mi) [0x00076] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Widgets.Widget.HandleMouseInputOuter (OpenRA.MouseInput mi) [0x000a0] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Widgets.Ui.HandleInput (OpenRA.MouseInput mi) [0x0001e] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.DefaultInputHandler+<>c__DisplayClass5_0.<OnMouseInput>b__0 () [0x00000] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Sync.RunUnsynced[T] (System.Boolean checkSyncHash, OpenRA.World world, System.Func`1[TResult] fn) [0x00023] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.DefaultInputHandler.OnMouseInput (OpenRA.MouseInput input) [0x0000d] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Platforms.Default.Sdl2Input.PumpInput (OpenRA.Platforms.Default.Sdl2PlatformWindow device, OpenRA.IInputHandler inputHandler, System.Nullable`1[T] lockedMousePosition) [0x00229] in <22c65db8b0cd40588acf8aad5ed77c2a>:0 
  at OpenRA.Platforms.Default.Sdl2PlatformWindow.PumpInput (OpenRA.IInputHandler inputHandler) [0x00006] in <22c65db8b0cd40588acf8aad5ed77c2a>:0 
  at OpenRA.Renderer.EndFrame (OpenRA.IInputHandler inputHandler) [0x00099] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Game.RenderTick () [0x001ba] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Game.Loop () [0x0015a] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Game.Run () [0x0003c] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <00bd104476d3433fb3f1c0e5739b97ca>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in <00bd104476d3433fb3f1c0e5739b97ca>:0
```

**Review by obrakmann — COMMENTED** [review]
Confirmed that this works for skirmish games and missions, saving and loading savegames also still works.  I can't test multiplayer games, unfortunately.  Would be good if someone did so.

**Comment by ubitux** [comment]
It doesn't work for multiplayer games: no players are visible after the replay starts.

**Comment by Mailaender** [comment]
I get a crash when trying to watch a multiplayer replay now:

```
Exception of type `System.ArgumentOutOfRangeException`: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: startIndex
  at System.BitConverter.ToInt32 (System.Byte[] value, System.Int32 startIndex) [0x00010] in <5514b7b0ec654df1b74df5da6e884f74>:0 
  at OpenRA.Network.ReplayConnection..ctor (System.String replayFilename) [0x00085] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Game.JoinReplay (System.String replayFile) [0x00005] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Mods.Common.Widgets.Logic.ReplayBrowserLogic.WatchReplay () [0x0002a] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.Logic.ReplayBrowserLogic+<>c__DisplayClass14_0.<.ctor>b__3 () [0x00000] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.ButtonWidget.<.ctor>b__42_5 (OpenRA.MouseInput _) [0x00000] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Mods.Common.Widgets.ButtonWidget.HandleMouseInput (OpenRA.MouseInput mi) [0x00076] in <72b9b463eb9b492c94140ac77e6b7147>:0 
  at OpenRA.Widgets.Widget.HandleMouseInputOuter (OpenRA.MouseInput mi) [0x000a0] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Widgets.Ui.HandleInput (OpenRA.MouseInput mi) [0x0001e] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.DefaultInputHandler+<>c__DisplayClass5_0.<OnMouseInput>b__0 () [0x00000] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Sync.RunUnsynced[T] (System.Boolean checkSyncHash, OpenRA.World world, System.Func`1[TResult] fn) [0x00023] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.DefaultInputHandler.OnMouseInput (OpenRA.MouseInput input) [0x0000d] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Platforms.Default.Sdl2Input.PumpInput (OpenRA.Platforms.Default.Sdl2PlatformWindow device, OpenRA.IInputHandler inputHandler, System.Nullable`1[T] lockedMousePosition) [0x00229] in <22c65db8b0cd40588acf8aad5ed77c2a>:0 
  at OpenRA.Platforms.Default.Sdl2PlatformWindow.PumpInput (OpenRA.IInputHandler inputHandler) [0x00006] in <22c65db8b0cd40588acf8aad5ed77c2a>:0 
  at OpenRA.Renderer.EndFrame (OpenRA.IInputHandler inputHandler) [0x00099] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Game.RenderTick () [0x001ba] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Game.Loop () [0x0015a] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Game.Run () [0x0003c] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <f7c2039239e04a90bfd2555605a70d4f>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in <f7c2039239e04a90bfd2555605a70d4f>:0
```

**Comment by Mailaender** [comment]
I propose to remove the breaking changes completely instead of trying to patch them up: https://github.com/OpenRA/OpenRA/pull/18424

**Pull request closed without merging**